### PR TITLE
Carapace Boss - Code

### PR DESCRIPTION
**Pull request by TideSofDarK**

### CODE REVIEW

**Comment by l34Um1** [comment]
This constantly crashes when I try to test it. Maybe a precaching issue?

**Comment by l34Um1** [comment]
Sick. Ready to merge imo.

File: game/scripts/vscripts/abilities/carapace/boss_carapace_headbutt.lua [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+		caster:EmitSound("Hero_Juggernaut.BladeDance")
+
+		local smoke = ParticleManager:CreateParticle("particles/econ/items/antimage/antimage_ti7_golden/antimage_blink_start_ti7_golden_smoke.vpcf", PATTACH_CUSTOMORIGIN, caster)
+		ParticleManager:SetParticleControl(smoke, 0, caster:GetAbsOrigin())
+		ParticleManager:ReleaseParticleIndex(smoke)
+
+		local enemies = self:GetEnemies()
+
+		for k,v in pairs(enemies) do
```
Comment (carlosrpg) [inline game/scripts/vscripts/abilities/carapace/boss_carapace_headbutt.lua]: avoid using one letter variables and don't name if you are not using, try something more meaningful like:
`for _,victim in pair(enemies) do`

File: game/scripts/vscripts/units/carapace.lua [hunk 2]
```diff
@@ -0,0 +1,104 @@
+function Spawn( entityKeyValues )
```
Comment (carlosrpg) [inline game/scripts/vscripts/units/carapace.lua]: try to follow the naming convention with ai_ prefix for AI files

File: game/scripts/npc/units/carapace/npc_dota_boss_carapace.txt [hunk 3]
```diff
@@ -0,0 +1,118 @@
+"DOTAUnits"
+{
+  //=================================================================================
+  // Boss tier ?
```
Comment (carlosrpg) [inline game/scripts/npc/units/carapace/npc_dota_boss_carapace.txt]: Tier 2

**Comment by TideSofDarK** [comment]
Should be fine as well

File: game/scripts/vscripts/units/ai_carapace.lua [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+		thisEntity:SetAcquisitionRange(0)
+	end
+
+	local enemies = FindUnitsInRadius(
+		thisEntity:GetTeamNumber(),
+		thisEntity:GetOrigin(), nil,
+		1000,
+		DOTA_UNIT_TARGET_TEAM_ENEMY,
+		DOTA_UNIT_TARGET_HERO,
+		DOTA_UNIT_TARGET_FLAG_MAGIC_IMMUNE_ENEMIES + DOTA_UNIT_TARGET_FLAG_FOW_VISIBLE,
```
Comment (carlosrpg) [inline game/scripts/vscripts/units/ai_carapace.lua]: the boss should target invisible units right?

**Pull request merged**